src/prefect/cli/run.py

```diff
@@ -140,16 +140,16 @@ def flow(
         $ prefect run flow -n "Test-Flow" -p "My Project" -ps '{"my_param": 42}'
         Flow Run: https://cloud.prefect.io/myslug/flow-run/2ba3rrfd-411c-4d99-bb2a-f64a6dea78f9
     """
-    if not id and not (name and project):
+    if not id and not (name and project) and not version_group_id:
         click.secho(
-            "A flow ID or some combination of flow name and project must be provided.",
+            "A flow ID, version group ID,  or some combination of flow name and project must be provided.",
             fg="red",
         )
         return
 
-    if id and (name or project):
+    if (id or version_group_id) and (name or project):
```

Comment:
This will still let both `id` and `version_group_id` be passed, when we probably only want to accept one or the other.